src/Util/SupportedAddonsLoader.php

```diff
@@ -2,21 +2,40 @@
 
 namespace BringYourOwnIdeas\Maintenance\Util;
 
+use BringYourOwnIdeas\Maintenance\Tasks\UpdatePackageInfoTask;
+use SilverStripe\Dev\Deprecation;
+
 /**
- * Handles fetching supported addon details from addons.silverstripe.org
+ * Handles fetching supported addon details from silverstripe/supported-modules
+ * @deprecated 3.2.0 Use BringYourOwnIdeas\Maintenance\Tasks\UpdatePackageInfoTask::getSupportedPackages() instead.
  */
 class SupportedAddonsLoader extends ApiLoader
 {
+    public function __construct()
+    {
+        Deprecation::withNoReplacement(
+            fn() => Deprecation::notice(
+                '3.2.0',
+                'Use ' . UpdatePackageInfoTask::class . '::getSupportedPackages() instead.',
+                Deprecation::SCOPE_CLASS
+            )
+        );
+    }
+
     /**
      * Return the list of supported modules
      *
      * @return array
      */
     public function getAddonNames()
     {
-        $endpoint = 'https://raw.githubusercontent.com/silverstripe/supported-modules/5/modules.json';
+        // Check for a cached value and return if one is available
+        $endpoint = 'https://raw.githubusercontent.com/silverstripe/supported-modules/main/repositories.json';
         return $this->doRequest($endpoint, function ($responseJson) {
-            return array_map(fn(array $item) => $item['composer'], $responseJson);
+            return array_filter(array_map(
+                fn(array $item) => isset($item['majorVersionMapping'][5]) ? $item['packagist'] : null,
```

Comment:
Didn't bother making the cms major here dynamic since this is deprecated anyway.